pkg/collect/logs_test.go

```diff
@@ -63,19 +74,26 @@ func Test_setLogLimits(t *testing.T) {
 			actual := corev1.PodLogOptions{}
 			setLogLimits(&actual, test.limits, convertMaxAgeToTime)
 
-			if test.expected.TailLines != nil {
-				req.NotNil(actual.TailLines)
-				assert.Equal(t, *test.expected.TailLines, *actual.TailLines)
+			if test.expected.LimitBytes != nil {
+				req.NotNil(actual.LimitBytes)
```

Comment:
While here, could you please change using `require` to `assert`? The former is meant to ensure a mandatory precondition is set, not assert a test case. In practice, if `require` were to fail, the test run stops at that point, which we would not want. The preference is to run through all test assertions and report all failures.

Reply:
Thats indeed exactly what we talked about earlier, thanks @banjoh. So you would ask yourself when would a test stop, after the 1st, 2nd or 3rd? which is something you indeed do not want. We do want all tests to be ran through.

Next steps would then be to rewrite using `assert..` instead of the currently specified `req..` and make sure we have a proper function testset for Troubleshoot logs.